caddyfile.go

```diff
@@ -140,6 +140,84 @@ func parseCaddyfileAuthPortal(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigVal
 					return nil, h.Errf("auth backend %s directive failed to compile to JSON: %s", rootDirective, err.Error())
 				}
 				portal.Backends = append(portal.Backends, backend)
+			case "rolemapping":
```

Comment:
@SvenDowideit , I get the idea here. How could you improve the code such that you minimize having duplicate code, i.e. the code in `backends`? Perhaps a separate function?

Reply:
yeah, I'm working towards adding external json files, and want to leave space for storing the email->role info in a database too
which is very quickly leading me to extracting it all to extracting it to its own module

Reply:
duplication gone :)